src/utils/fluxNode.ts

```diff
@@ -197,8 +198,28 @@ export function appendTextToFluxNodeAsGPT(
 
     const copy = { ...node, data: { ...node.data } };
 
+    // To reduce complexity we assume auto-labels either end with " ..." or are equal to the generated text.
+    const isTruncated = copy.data.label.endsWith(" ...");
+    const isPreviousAutoLabel = isTruncated || copy.data.label === copy.data.text;
+    const isNewPrompt = copy.data.text.length === 0;
+
     copy.data.text += text;
 
+    // Preserve custom labels
+    if (
+      copy.data.label !== displayNameFromFluxNodeType(FluxNodeType.GPT) &&
+      !isPreviousAutoLabel
+    )
+      return copy;
+
+    // If label hasn't reached max length or it's a new prompt, set from text.
+    // Once label reaches max length, truncate it.
+    if (copy.data.label.length < MAX_AUTOLABEL_LENGTH || isNewPrompt) {
+      copy.data.label = copy.data.text;
+    } else if (!isTruncated) {
+      copy.data.label += " ...";
+    }
```

Comment:
wait are we just getting lucky or does this not snap to words? arent we getting tokens streamed in which could be parital words

Reply:
yeah I know, but I must have generated like a hundred nodes while doing this PR and never got a half word. that's why I didn't add the check here

Reply:
🤷 haha welp if it works it works. can address w/ a future PR if we ever see weirdness happen in practice 👍